scipy/stats/morestats.py

```diff
@@ -1663,7 +1674,55 @@ def shapiro(x):
     0.08144091814756393
 
     """
-    x = np.ravel(x)
+    x = np.ma.asarray(x)
+
+    # check x size
+    if x.size == 0:
+        raise ValueError("Empty array is given.")
+
+    if axis is None:
```

Comment:
so here we should emit a deprecation warning,

something like 

```
        warnings.warn(
            "the default behavior of axis=None in shapiro will change to axis=0 in SciPy version 1.8.0",
            DeprecationWarning
        )
```

AFAICS all other uses of `axis` argument come after this one, and may appear in private functions so they are fine as written.

Reply:
I looked throughout github and it doesn't look like there are more than 1 repository coming up that invoke the `shapiro()` method and there it appears as an ad-hoc script with use of a list of numbers as input. I'm going to propose to call this improving implementation details rather than what I said 45 minutes ago, in other words we do not go through the deprecation warning process and instead go directly to the `axis=0` approach. If we hear anything between now and the next release we can revert. 

If we went the previous route that I was suggesting w/the deprecation warning, we'd be emitting a warning on every invocation of the `shapiro()` call.